ouroboros-network/src/Ouroboros/Network/NodeToClient.hs

```diff
@@ -231,3 +246,108 @@ withServer_V1 muxTracer handshakeTracer tbl addr peeridFn versionData applicatio
         versionData
         (DictVersion nodeToClientCodecCBORTerm)
         application)
+
+-- | 'ncSubscriptionWorker' which starts given application versions on each
+-- established connection.
+--
+ncSubscriptionWorker
+    :: forall appType peerid void x y.
+       ( HasInitiator appType ~ True
+       , Show peerid )
+    => Tracer IO (WithIPList (SubscriptionTrace Socket.SockAddr))
+    -> Tracer IO (WithMuxBearer (MuxTrace NodeToClientProtocols))
+    -> Tracer IO
+        (TraceSendRecv
+          (Handshake NodeToClientVersion CBOR.Term)
+          peerid
+          (DecoderFailureOrTooMuchInput DeserialiseFailure))
+    -> (Socket.SockAddr -> Socket.SockAddr -> peerid)
+    -> ConnectionTable IO Socket.SockAddr
+    -> LocalAddresses Socket.SockAddr
+    -> (Socket.SockAddr -> Maybe DiffTime)
+    -- ^ Lookup function, should return expected delay for the given address
+    -> IPSubscriptionTarget
```

Comment:
And for the local case do we not want to specialise this too? There's only one (local socket) target, and valency of 1.

Reply:
I don't think we should specialize this. A client could be talking to more then one node for redundancy reasons.